lib/spack/llnl/util/filesystem.py

```diff
@@ -226,6 +226,12 @@ def mime_type(filename):
     type, _, subtype = output.partition("/")
     return type, subtype
 
+def binary_mime_type(filename):
```

Comment:
I'm not sure what the term `binary_` is supposed to add to the name (i.e. what it adds to `mime_type`). Calling this `mime_type` and renaming the existing `mime_type` to `file_mime_type` (or `unix_mime_type`) makes more sense to me.

Reply:
IIRC my thought was that this was only dealing with binaries as opposed to all file types like `file` normally does. Agree with your suggestion, will implement.